YoCode/RunParameterChecker.cs

```diff
@@ -20,25 +19,63 @@ internal class RunParameterChecker : IRunParameterChecker
         public string AppCodeBaseCost { get; set; }
         public string AppDuplicationCost { get; set; }
 
-        public RunParameterChecker(Output compositeOutput, IInputResult result, IAppSettingsBuilder appsettingsBuilder)
+        public RunParameterChecker(Output compositeOutput, IAppSettingsBuilder appsettingsBuilder)
         {
             this.compositeOutput = compositeOutput;
-            this.result = result;
             this.appsettingsBuilder = appsettingsBuilder;
         }
 
-        public bool ParametersAreValid()
+        public bool ParametersAreValid(string outputPath, IInputResult result)
+        {
+            CheckInput(outputPath, result);
+
+            ReadAppsettings();
+
+            CheckIfAppsettingsValuesAreValid();
+
+            return CheckIfToolExecutablesExist();
+        }
+
+        private void CheckIfAppsettingsValuesAreValid()
+        {
+            var CMDPathExists = CheckToolDirectory(CMDToolsPath, "CMDtoolsDir");
+            var dotCoverPathExists = CheckToolDirectory(DotCoverDir, "dotCoverDir");
+            var costValuesProvided = CheckIfCostsProvided(TestCodeBaseCost, TestDuplicationCost, "Test cost values")
+                && CheckIfCostsProvided(AppCodeBaseCost, AppDuplicationCost, "App cost values");
+
+            var juniorFileExists = FileExists("JuniorWeightings.json");
+            var originalFileExists = FileExists("OriginalWeightings.json");
+
+            bool anyFilesMissing = !CMDPathExists || !dotCoverPathExists || !juniorFileExists || !originalFileExists || !costValuesProvided;
+
+            if (anyFilesMissing)
+            {
+                compositeOutput.ShowInputErrors(Errs);
+                Environment.Exit(1);
+            }
+        }
+
+        private void CheckInput(string outputPath, IInputResult result)
         {
             if (result.HelpAsked)
             {
-                return false;
+                compositeOutput.ShowHelp();
+                if (result.CreateHtmlReport && result.OpenHtmlReport)
+                {
+                    WebWriter.LaunchReport(result, outputPath);
+                }
```

Comment:
I don't think all this ShowHelp and LaunchReport logic should be in this class. It should check the parameters but it shouldn't be the thing that controls what actions are taken.